lib/pure/uri.nim

```diff
@@ -483,7 +483,11 @@ func `?`*(u: Uri, query: openArray[(string, string)]): Uri =
     let foo = parseUri("https://example.com") / "foo" ? {"bar": "qux"}
     assert $foo == "https://example.com/foo?bar=qux"
   result = u
-  result.query = encodeQuery(query)
+  # Makes sure to keep previous query if found
+  if result.query != "":
+    result.query = result.query & "&" & encodeQuery(query)
```

Comment:
Why make a new string instead of just `add`ing the & and encodeQuery(query) ?

Reply:
```suggestion
    result.query.add '&'
    result.query.add encodeQuery(query)
```

Something like this (of course you can do this in 1 line as well)

Reply:
Nothing wrong with using add instead, I just sort of forgot about it as I wanted to make it work first before worrying about optimizations. I guess python habits got in the way.

Reply:
I was curious and ran some benchmarks real quick and my results suggest that the original method via setting is more than 2x times faster than using `and`. [Playground benchmark](https://play.nim-lang.org/#ix=3Y3n) 
Unless there is a different reason to you what you suggested, I think I will revert it to how it was originally.

Reply:
@Nmerryman I also ran some benchmarks with benchy, see this:
```nim
import std/uri

# presetting
# for a in 0 ..< runs:
#   adding[a].query.add("a")
#   setting[a].query = "a"

import benchy

timeIt "adding":
  var q = initUri()
  for i in 0..1000:
    q.query.add("aaaaaaaaaaa")
    q.query.add("aaaaaaaaaaa")
    keep q

timeIt "setting":
  var q = initUri()
  for i in 0..1000:
    q.query = q.query & "aaaaaaaaaaa" & "aaaaaaaaaaa"
    keep q
```

here your method is much slower

Reply:
Anyway, my comment was merely a suggestion, if you consider your way to be better, go ahead with it.

Reply:
Also you can check the memory allocation usage:
```nim
import std/uri

when true:
  var q = initUri()
  for i in 0..1000:
    q.query.add("aaaaaaaaaaa")
    q.query.add("aaaaaaaaaaa")
  doAssert q.query.len > 5000

when false:
  var q = initUri()
  for i in 0..1000:
    q.query = q.query & "aaaaaaaaaaa" & "aaaaaaaaaaa"
  doAssert q.query.len > 5000
```
  
When the first branch is true we get this (with `-d:danger --gc:arc -d:useMalloc):
`total heap usage: 12 allocs, 12 frees, 45,153 bytes allocated`

In second case we get:
`total heap usage: 1,001 allocs, 1,001 frees, 11,042,031 bytes allocated`